src/test/java/software/amazon/smithy/lsp/language/InlayHintHandlerTest.java

```diff
@@ -0,0 +1,350 @@
+    }
+
+    @Test
+    public void inlayHintForInlineOperationOffRangeSuffix() {
+        TextWithPositions model = TextWithPositions.from("""
+                $version: "2"
+                $operationInputSuffix: "Request"
+                $operationOutputSuffix: "Response"
+                
+                namespace smithy.example
+                
+                %operation GetUser {
+                    input :=% {
+                        userId: String
+                    }
+                
+                    output :=% {
+                        username: String
+                        userId: String
+                    }
+                }
+                %
+                """);
+        var positions = model.positions();
+        Position startPosition = new Position(positions[0].getLine(),0);
+        Position endPosition = new Position(positions[3].getLine(),0);
+        List<InlayHint>hints = getInlayHints(model.text(), startPosition, endPosition);
+        assertThat(hints, hasSize(2));
+        assertEquals(new Position(positions[1].getLine(), positions[1].getCharacter()), hints.get(0).getPosition());
+        assertEquals("GetUserRequest", hints.get(0).getLabel().getLeft());
+        assertEquals(new Position(positions[2].getLine(), positions[2].getCharacter()), hints.get(1).getPosition());
+        assertEquals("GetUserResponse", hints.get(1).getLabel().getLeft());
+    }
+
+    @Test
+    public void inlayHintForInlineOperationPartiallyInRange() {
+        TextWithPositions model = TextWithPositions.from("""
+                $version: "2"
+                $operationInputSuffix: "Request"
+                $operationOutputSuffix: "Response"
+                
+                namespace smithy.example
+                
+                %operation GetUser {
+                    input :=% {
+                        userId: String
+                    }
+                %
+                    output := {
+                        username: String
+                        userId: String
+                    }
+                }
+                """);
+        var positions = model.positions();
+        Position startPosition = new Position(positions[0].getLine(),0);
+        Position endPosition = new Position(positions[2].getLine(),0);
+        List<InlayHint>hints = getInlayHints(model.text(), startPosition, endPosition);
+        assertThat(hints, hasSize(1));
+        assertEquals(new Position(positions[1].getLine(), positions[1].getCharacter()), hints.get(0).getPosition());
+        assertEquals("GetUserRequest", hints.get(0).getLabel().getLeft());
+    }
+
+    @Test
+    public void inlayHintForInlineOperationNotInRange() {
+        TextWithPositions model = TextWithPositions.from("""
+                $version: "2"
+                
+                operation GetUser {
+                    input := {
+                        userId: String
+                    }
+                
+                %    output := {
```

Comment:
```suggestion
                %    output :=% {
```
to check the hint position. I'd just do this on all the tests - there's some tests throughout the codebase where I didn't check the position, but I think that was a mistake because it's less resilient to changes down the road.